homeassistant/components/myuplink/number.py

```diff
@@ -0,0 +1,132 @@
+    "NIBEF": {
+        "40940": NumberEntityDescription(
+            key="degree_minutes",
+            icon="mdi:thermometer-lines",
+            native_unit_of_measurement="DM",
+        ),
+    },
+}
+
+
+def get_description(device_point: DevicePoint) -> NumberEntityDescription | None:
+    """Get description for a device point.
+
+    Priorities:
+    1. Category specific prefix e.g "NIBEF"
+    2. Global parameter_unit e.g. "DM"
+    3. Default to None
+    """
+    prefix, _, _ = device_point.category.partition(" ")
+    description = CATEGORY_BASED_DESCRIPTIONS.get(prefix, {}).get(
+        device_point.parameter_id
+    )
+
+    if description is None:
+        description = DEVICE_POINT_UNIT_DESCRIPTIONS.get(device_point.parameter_unit)
+
+    return description
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up myUplink number."""
+    entities: list[NumberEntity] = []
+    coordinator: MyUplinkDataCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    # Setup device point number entities
+    for device_id, point_data in coordinator.data.points.items():
+        for point_id, device_point in point_data.items():
+            description = get_description(device_point)
+            if find_matching_platform(device_point, description) == Platform.NUMBER:
+                entities.append(
+                    MyUplinkNumber(
+                        coordinator=coordinator,
+                        device_id=device_id,
+                        device_point=device_point,
+                        entity_description=description,
+                        unique_id_suffix=point_id,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class MyUplinkNumber(MyUplinkEntity, NumberEntity):
+    """Representation of a myUplink number entity."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: NumberEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the number."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        # Internal properties
+        self.point_id = device_point.parameter_id
+        self._attr_name = device_point.parameter_name
+        self._attr_native_min_value = (
+            device_point.raw["minValue"] if device_point.raw["minValue"] else -30000
+        ) * float(device_point.raw.get("scaleValue", 1))
+        self._attr_native_max_value = (
+            device_point.raw["maxValue"] if device_point.raw["maxValue"] else 30000
+        ) * float(device_point.raw.get("scaleValue", 1))
+        self._attr_step_value = device_point.raw.get("stepValue", 20)
```

Comment:
I suppose we could move this up to `get_description` and then use it also in `find_matching_platform`

Reply:
The `name` could also go into `get_description`

Reply:
Definitely possible, but we chose to do it way in previous platforms (sensor, binary_sensor, switch) following advice from the reviewer. The EntityDescription classes are immutable and that makes it more complex to change or add attributes. We pick up static data from get_description and then add other more dynamic attributes by setting \_attr\_*  this way.